chain-storage/src/iterator.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{permanent_store::PermanentStore, BlockInfo, Error, Value};
+use sled::Tree;
+
+/// Iterator over blocks. Starts from n-th ancestor of the given block.
+pub struct StorageIterator {
+    state: IteratorState,
+    to: Value,
+    block_info: Tree,
+    blocks: Tree,
+}
+
+enum IteratorState {
+    Permanent {
+        iter: data_pile::SeqNoIter,
+        current_length: u32,
+        stop_at_length: u32,
+    },
+    Volatile {
+        ids: Vec<Value>,
+    },
+}
+
+impl StorageIterator {
+    /// Create a new iterator. `from` and `to` values MUST have a path between
+    /// them and this needs to be checked before the iterator is started.
```

Comment:
How is this MUST enforced?
Please detail if it is a panic, by using a conventional `# Panics` section.

Reply:
Well, this actually needs to be updated, this is an outdated piece of documentation. There is no `from` supplied to `Iterator::new`.